### ISSUE TITLE
solver_describe_decision()

### ISSUE THREAD

**Issue by mluscon** [openSUSE/libsolv#130]
Hi Michael,

thank you for your previous fix. Unfortunately, I found another issue in solver_describe_decision()  reproducible by:
```
repo system 0 testtags <inline>
repo available 0 testtags <inline>
#>=Pkg: A 1 1 x86_64
#>=Rec: B >= 1-0
#>=Pkg: B 1 1 noarch
system x86_64 * system
poolflags implicitobsoleteusescolors
solverflags allowvendorchange keepexplicitobsoletes bestobeypolicy keeporphans yumobsoletes
job install pkg A-1-1.x86_64
job multiversion provides kernel
job multiversion provides installonlypkg(kernel)
job multiversion provides installonlypkg(kernel-module)
job multiversion provides installonlypkg(vm)
result transaction,recommended
```

**Comment by mlschroe** [comment]
Well, I don't see a bug. Are you talking about the reason for the installation of A being a unit rule? That's correct, as there's just one package providing A the solver sees a unit rule. The rule itself is a job rule.

Basically you can't use describe_decision to find out if a package was installed by user request, as the package may also get installed because of a package dependency before the solver resolves the particular user request. That's why solver_get_userinstalled does not look at the reason, but just checks for packages matched by a job.

**Comment by mluscon** [comment]
Correct me if I am wrong, I suppose the reason of package B in the transaction should be SOLVER_REASON_WEAKDEP as it was pulled in by Recommends relation.

**Comment by mlschroe** [comment]
But I do get SOLVER_REASON_WEAKDEP for B. What does it return in your test?
```
reason A-1-1.x86_64@available unit 300db6ce502dde94261e267a8c535441
reason B-1-1.noarch@available recommended B >= 1-0 A-1-1.x86_64@available
```

**Comment by mluscon** [comment]
Oh, I made a typo in my testing code. I am sorry for bothering you.
